pkg/config/config.go

```diff
@@ -32,6 +33,22 @@ type ComponentSettings struct {
 	// The builder image to use
 	ComponentType *string `yaml:"ComponentType,omitempty"`
 
+	// Path is path to binary in current/context dir
+	Path *string `yaml:"Path,omitempty"`
+
+	// Ref is component source git ref but can be levaraged for more in future
+	Ref *string `yaml:"Ref,omitempty"`
+
+	// Type is type of component source: git/local/binary
+	Type *string `yaml:"Type,omitempty"`
+
+	// Ports is a slice of ports to be exposed when a component is created
+	Ports *string `yaml:"Ports,omitempty"`
+
+	App *string `yaml:"App,omitempty"`
+
+	Project *string `yaml:"Project,omitempty"`
+
 	ComponentName *string `yaml:"ComponentName,omitempty"`
```

Comment:
Could we use just Name?

Reply:
I think I'll keep its refactoring to another PR ;)
It was anyway not created by this PR

Reply:
> I think I'll keep its refactoring to another PR ;)
> It was anyway not created by this PR

OK, but please make note of this somewhere. Ideally, create an issue that will contain all that you want to address in another PR.

Reply:
Addressed as part of #1453